### PR TITLE
Prepared a Shared VPC HTTP ILB for inclusion in C.G.C. documentation

### PR DESCRIPTION
**Pull request by msampathkumar**
This snippet is prepared for user-request.

A TF will be create under https://cloud.google.com/vpc/docs/shared-vpc

```release-note:none
```

### CODE REVIEW

**Review by bharathkkb — COMMENTED** [review]
Some minor comments and I am new to samples so deferring to @rogerthatdev or @betsy-lichtenberg for another look.

File: shared_vpc_http_ilb_with_mig_backend/main.tf [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+resource "google_project_iam_binding" "default" {
+  project = "my-host-project"
+  role    = "roles/compute.networkUser"
+
+  members = [
+    "serviceAccount:${data.google_project.service_project02.number}@cloudservices.gserviceaccount.com",
+  ]
+}
+
+# test instance
```
[the remark this answers is not in the document]
Comment (msampathkumar) [reply shared_vpc_http_ilb_with_mig_backend/main.tf:217]: Hi Bharath, so this instance is used for testing. Testing steps are recorded there http://b/232285286#comment15.

This is similar to https://cloud.google.com/load-balancing/docs/l7-internal/l7-internal-shared-vpc#testing

File: shared_vpc_http_ilb_with_mig_backend/main.tf [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      Name: $NAME
+      IP: $IP
+      Metadata: $METADATA
+      </pre>
+      EOF
+    EOF1
+  }
+  lifecycle {
+    create_before_destroy = true
+  }
```
[the remark this answers is not in the document]
Comment (msampathkumar) [reply shared_vpc_http_ilb_with_mig_backend/main.tf:181]: Hi, I did not understand. Are you referring to `lifecycle` attribute?

[the remark this answers is not in the document]
Comment (bharathkkb) [reply shared_vpc_http_ilb_with_mig_backend/main.tf:181]: Correct, I was wondering what was the need for create before destroy.

File: shared_vpc_http_ilb_with_mig_backend/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (bharathkkb) [reply shared_vpc_http_ilb_with_mig_backend/main.tf:217]: Makes sense, thanks for clarifying!

File: shared_vpc_http_ilb_with_mig_backend/main.tf [hunk 3]
```diff
@@ -0,0 +1,240 @@
+# Shared VPC Internal HTTP load balancer with a managed instance group backend
+
+# [START cloudloadbalancing_shared_vpc_int_http_gce]
+# VPC network
+resource "google_compute_network" "default" {
+  name                    = "l7-ilb-network"
+  provider                = google
```
[the remark this answers is not in the document]
Comment (msampathkumar) [reply shared_vpc_http_ilb_with_mig_backend/main.tf:7]: Thank for the clarity. I will remove `provider` attribute as it would be same as default.

**Review by bharathkkb — APPROVED** [review]

**Comment by msampathkumar** [comment]
Thank you @bharathkkb

**Pull request merged**